src/methods/scprint/script.py

```diff
@@ -0,0 +1,215 @@
+    "input_test": "resources_test/task_label_projection/cxg_immune_cell_atlas/test.h5ad",
+    "output": "output.h5ad",
+    "model_name": "large",
+    "model": None,
+}
+meta = {"name": "scprint"}
+## VIASH END
+
+print(f"====== scPRINT version {scprint.__version__} ======", flush=True)
+
+print("\n>>> Reading input data...", flush=True)
+input_train = ad.read_h5ad(par['input_train'])
+input_test = ad.read_h5ad(par['input_test'])
+
+if input_train.uns["dataset_organism"] == "homo_sapiens":
+    input_train.obs["organism_ontology_term_id"] = "NCBITaxon:9606"
+    input_test.obs["organism_ontology_term_id"] = "NCBITaxon:9606"
+elif input_train.uns["dataset_organism"] == "mus_musculus":
+    input_train.obs["organism_ontology_term_id"] = "NCBITaxon:10090"
+    input_test.obs["organism_ontology_term_id"] = "NCBITaxon:10090"
+else:
+    raise ValueError(
+        f"scPRINT requires human or mouse data, not '{input.uns['dataset_organism']}'"
+    )
+
+adata = input_train.copy()
+adata.X = adata.layers["counts"]
+
+adata_test = input_test.copy()
+adata_test.X = adata_test.layers["counts"]
+
+print("\n>>> Preprocessing train data...", flush=True)
+preprocessor = Preprocessor(
+    # Lower this threshold for test datasets
+    min_valid_genes_id=1000 if input_train.n_vars < 2000 else 10000,
+    # Turn off cell filtering to return results for all cells
+    filter_cell_by_counts=False,
+    min_nnz_genes=False,
+    do_postp=False,
+    # Skip ontology checks
+    skip_validate=True,
+)
+adata = preprocessor(adata)
+
+model_checkpoint_file = par["model"]
+if model_checkpoint_file is None:
+    print(f"\n>>> Downloading '{par['model_name']}' model...", flush=True)
+    model_checkpoint_file = hf_hub_download(
+        repo_id="jkobject/scPRINT", filename=f"{par['model_name']}.ckpt"
+    )
+
+print(f"Model checkpoint file: '{model_checkpoint_file}'", flush=True)
+model = scPrint.load_from_checkpoint(
+    model_checkpoint_file,
+    transformer="normal",  # Don't use this for GPUs with flashattention
+    precpt_gene_emb=None,
+)
+
+print("\n>>> Embedding train data...", flush=True)
+if torch.cuda.is_available():
+    print("CUDA is available, using GPU", flush=True)
+    precision = "16"
+    dtype = torch.float16
+else:
+    print("CUDA is not available, using CPU", flush=True)
+    precision = "32"
+    dtype = torch.float32
+
+n_cores_available = len(os.sched_getaffinity(0))
+
+print(f"Using {n_cores_available} worker cores")
+embedder = Embedder(
+    batch_size=32,
+    how="random expr",
+    max_len=4000,
+    add_zero_genes=0,
+    num_workers=n_cores_available,
+    doclass=True,
+    doplot=False,
+    precision=precision,
+    dtype=dtype,
+    pred_embedding=["cell_type_ontology_term_id"],
+    keep_all_cls_pred=False,
+    output_expression="none"
+)
+embedded, _ = embedder(model, adata, cache=False)
+
+print("\n>>> Matching predictions to labels...", flush=True)
```

Comment:
Maybe we should move this to a helper function so it can be reused across methods?

Reply:
Yes, that would be good I guess. Should I create a new PR for it?